tests/nosetests/pyanaconda_tests/module_source_nfs_test.py

```diff
@@ -0,0 +1,177 @@
+#
+# Copyright (C) 2020  Red Hat, Inc.
+#
+# This copyrighted material is made available to anyone wishing to use,
+# modify, copy, or redistribute it subject to the terms and conditions of
+# the GNU General Public License v.2, or (at your option) any later version.
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY expressed or implied, including the implied warranties of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General
+# Public License for more details.  You should have received a copy of the
+# GNU General Public License along with this program; if not, write to the
+# Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA
+# 02110-1301, USA.  Any Red Hat trademarks that are incorporated in the
+# source code or documentation are not subject to the GNU General Public
+# License and may only be used or replicated with the express permission of
+# Red Hat, Inc.
+#
+import unittest
+from unittest.mock import patch
+
+from pyanaconda.core.constants import INSTALL_TREE
+from pyanaconda.modules.common.constants.interfaces import PAYLOAD_SOURCE_NFS
+from pyanaconda.modules.payloads.constants import SourceType
+from pyanaconda.modules.payloads.source.nfs.nfs import NFSSourceModule
+from pyanaconda.modules.payloads.source.nfs.nfs_interface import NFSSourceInterface
+from pyanaconda.modules.payloads.source.nfs.initialization import SetUpNFSSourceTask, \
+    TearDownNFSSourceTask
+from pyanaconda.payload.utils import PayloadSetupError
+
+from tests.nosetests.pyanaconda_tests import PropertiesChangedCallback
+
+
+nfs_address = "example.com:/some/path"
+nfs_url = "nfs:" + nfs_address
+mount_location = "/mnt/put-nfs-here"
+
+
+class NFSSourceInterfaceTestCase(unittest.TestCase):
+
+    def setUp(self):
+        self.source_module = NFSSourceModule()
+        self.source_interface = NFSSourceInterface(self.source_module)
+
+        self.callback = PropertiesChangedCallback()
+        self.source_interface.PropertiesChanged.connect(self.callback)
+
+    def type_test(self):
+        """Test NFS source has a correct type specified."""
+        self.assertEqual(SourceType.NFS.value, self.source_interface.Type)
+
+    def url_empty_properties_test(self):
+        """Test NFS source URL property when not set."""
+        self.assertEqual(self.source_interface.URL, "")
+
+    def url_properties_test(self):
+        """Test NFS source URL property is correctly set."""
+        self.source_interface.SetURL(nfs_url)
+        self.assertEqual(self.source_interface.URL, nfs_url)
+        self.callback.assert_called_once_with(
+            PAYLOAD_SOURCE_NFS.interface_name, {"URL": nfs_url}, [])
```

Comment:
Please use the `check_dbus_property()` method. See other tests for inspiration how to use it.

Reply:
However, this one seems valid to me.

Reply:
Done!